### ISSUE TITLE
tabLength setting not respected

### ISSUE THREAD

**Issue by danielchatfield** [atom/atom#8394]
I have set `tabLength` to 2 for `jsx` files via the settings page and can confirm that it is in the config.cson file but the `tabLength` actually used is 4.

You can see a video here: <http://uploads.vpcdn.co/1dVjv+>

**Comment by izuzak** [comment]
@danielchatfield I'm probably misunderstanding something in that video. As far as I can tell, the file which you show (`MerchantEnricher`) uses 2 spaces for a tab, which matches the config settings. Can you clarify where exactly you see that 4 spaces are used for a tab in that file? Also, can you share the file in question, e.g. in a gist?

**Comment by pwaller** [comment]
I wonder if you're hitting the problem I was just hitting.

I would change the language-specific tab length to 2 for javascript, and then it would mysteriously change itself back to 8 a few seconds later. Curiously, it did not change it back if I selected tab lengths 3 or 5.

I managed to make it do what I needed by removing my global tabLength setting, leaving that as the default. Then my language-specific tab lengths were respected correctly.
